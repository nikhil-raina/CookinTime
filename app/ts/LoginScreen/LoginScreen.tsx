/* eslint-disable prettier/prettier */
/* eslint-disable react-native/no-inline-styles */
/**
 * @providesModule LoginScreen
 * @flow
 */

import React from 'react';
import styles from './LoginScreenStyle';
import images from '../CIT-Modules/ImageTree';

import {
  View,
  Image,
  Text,
  ImageBackground,
} from 'react-native';

export default class LoginScreen extends React.Component<{}, {}> {

  constructor(props: any) {
    super(props);
  }

  render(): React.ReactElement {
    const background_image = images.LOGIN_BACKGROUND;
    const facebookLogo = images.LOGIN_FACEBOOK_LOGO;
    const facebookLogoShadow = images.LOGIN_FACEBOOK_LOGO_SHADOW;
    const emailLogo = images.LOGIN_EMAIL_LOGO;
    const emailLogoShadow = images.LOGIN_EMAIL_LOGO_SHADOW;
    const googleLogo = images.LOGIN_GOOGLE_LOGO;
    const googleLogoShadow = images.LOGIN_GOOGLE_LOGO_SHADOW;
    const headerDialogue = 'Kitchen awaits you!!';
    const firstLineMessage = 'To get a personalized experience, ';
    const secondLineMessage = 'please sign in with:';
    return (
      <ImageBackground  source={background_image}
                        style={styles.backgroundImage}>
        <View style={styles.container}>
          <Text style={styles.headerTextContainer}>
            {headerDialogue}
          </Text>
          <View style={styles.messageContainer}>
            <Text style={styles.inLineTextContainer}>
              {firstLineMessage}
            </Text>
            <Text style={styles.inLineTextContainer}>
              {secondLineMessage}
            </Text>
          </View>
          <View style={styles.logosContainer}>
            {/* TODO: Make these Logos clickable.
            Try using the Touchable Opacity. After clicking them, they need
            to go to the respective ways of logging into the app. */}
            <View style={styles.imageContainer}>
              <View style={styles.dropShadow}>
                <Image  source={facebookLogo}
                        style={styles.logoStyle}/>
              </View>
              <Text style={styles.logoTextContainer}>
                {'Facebook'}
              </Text>
            </View>
            <View style={styles.imageContainer}>
              <View style={styles.dropShadow}>
                <Image  source={googleLogo}
                        style={styles.logoStyle}/>
              </View>
              <Text style={styles.logoTextContainer}>
                {'Google'}
              </Text>
            </View>
            <View style={styles.imageContainer}>
              <View style={styles.dropShadow}>
                <Image  source={emailLogo}
                        style={styles.logoStyle}/>
              </View>
              <Text style={styles.logoTextContainer}>
                {'Email'}
              </Text>
            </View>
          </View>
          <View style={styles.termsAndConditionsContainer}>
            <Text style={styles.otherTextContainer}>
              {'By continuing, you accept our'}
            </Text>
            <View style={{flexDirection:'row'}}>
              <Text style={styles.termsAndConditionsTextContainer}>
                {/* TODO: Need to make this as a hyperlink that takes the user to the
                Document that goes about showing the Privacy Notices */}
                {'Privacy Notices'}
              </Text>
              <Text style={styles.otherTextContainer}>
                {' and '}
              </Text>
              <Text style={styles.termsAndConditionsTextContainer}>
                {/* TODO: Need to make this as a hyperlink that takes the user to the
                Document that goes about showing the Terms of Use */}
                {'Terms of Use'}
              </Text>
            </View>
          </View>
        </View>
      </ImageBackground>
    );
  }
}
