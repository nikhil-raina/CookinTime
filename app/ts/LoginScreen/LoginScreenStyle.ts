/* eslint-disable prettier/prettier */
/**
 * @provideModule LoginScreenStyle
 * @flow
 */

import {
  StyleSheet,
  Dimensions,
} from 'react-native';


export default StyleSheet.create({
  container: {
    alignItems: 'center',
  },

  backgroundImage: {
    height: Dimensions.get('window').height,
    width: Dimensions.get('window').width,
    backgroundColor: 'rgba(0,0,0,0.6)',

  },

  logosContainer: {
    width: 230,
    flexDirection: 'row',
    marginBottom: 50,
  },

  imageContainer: {
    alignItems: 'center',
    flex: 1,
  },

  logoTextContainer: {
    fontFamily: 'JosefinSans-Light',
    fontSize: 10,
    color: '#F2E6C2',
  },

  logoStyle: {
    height: 60,
    width: 60,
  },

  logoStyleShadow: {
    alignItems: 'center',
    height: 75,
    width: 75,
  },

  headerTextContainer: {
    paddingTop: 60,
    fontFamily: 'JosefinSans-Medium',
    fontSize: 34,
    color: '#F2E6C2',
    marginBottom: 60,
  },

  messageContainer: {
    alignItems: 'center',
    marginBottom: 30,
  },

  termsAndConditionsContainer: {
    alignItems: 'center',
  },

  otherTextContainer: {
    fontFamily: 'JosefinSans-Light',
    fontSize: 10,
    color: '#F2E6C2',
  },

  termsAndConditionsTextContainer: {
    fontFamily: 'JosefinSans-Light',
    fontSize: 10,
    color: '#F2E6C2',
    fontWeight: 'bold',
    textDecorationLine: 'underline',
  },

  inLineTextContainer: {
    fontFamily: 'JosefinSans-Light',
    fontSize: 18,
    color: '#F2E6C2',
  },
});
