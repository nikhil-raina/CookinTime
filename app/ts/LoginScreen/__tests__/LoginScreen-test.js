/* eslint-disable prettier/prettier */
import React from 'react';
import renderer from 'react-test-renderer';

import LoginScreen from '../LoginScreen';

describe('Rendering basic demo screen', () => {
  it('renders correctly', () => {
    const screen = renderer.create(<LoginScreen />);

    expect(screen.toJSON()).toMatchSnapshot();
  });
});
