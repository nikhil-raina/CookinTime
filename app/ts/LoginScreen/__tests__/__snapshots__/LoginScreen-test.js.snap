// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering basic demo screen renders correctly 1`] = `
<View
  accessibilityIgnoresInvertColors={true}
  style={
    Object {
      "backgroundColor": "rgba(0,0,0,0.6)",
      "height": 1334,
      "width": 750,
    }
  }
>
  <Image
    source={
      Object {
        "testUri": "../../../assets/images/LoginScreen/loginScreen-background.png",
      }
    }
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": 1334,
          "width": 750,
        },
        undefined,
      ]
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#F2E6C2",
          "fontFamily": "JosefinSans-Medium",
          "fontSize": 34,
          "marginBottom": 60,
          "paddingTop": 60,
        }
      }
    >
      Kitchen awaits you!!
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginBottom": 30,
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F2E6C2",
            "fontFamily": "JosefinSans-Light",
            "fontSize": 18,
          }
        }
      >
        To get a personalized experience, 
      </Text>
      <Text
        style={
          Object {
            "color": "#F2E6C2",
            "fontFamily": "JosefinSans-Light",
            "fontSize": 18,
          }
        }
      >
        please sign in with:
      </Text>
    </View>
    <View
      style={
        Object {
          "flexDirection": "row",
          "marginBottom": 50,
          "width": 230,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 5,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/images/LoginScreen/facebook-logo.png",
              }
            }
            style={
              Object {
                "height": 60,
                "width": 60,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2E6C2",
              "fontFamily": "JosefinSans-Light",
              "fontSize": 10,
            }
          }
        >
          Facebook
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 5,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/images/LoginScreen/google-logo.png",
              }
            }
            style={
              Object {
                "height": 60,
                "width": 60,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2E6C2",
              "fontFamily": "JosefinSans-Light",
              "fontSize": 10,
            }
          }
        >
          Google
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
          }
        }
      >
        <View
          style={
            Object {
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.8,
              "shadowRadius": 5,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/images/LoginScreen/email-logo.png",
              }
            }
            style={
              Object {
                "height": 60,
                "width": 60,
              }
            }
          />
        </View>
        <Text
          style={
            Object {
              "color": "#F2E6C2",
              "fontFamily": "JosefinSans-Light",
              "fontSize": 10,
            }
          }
        >
          Email
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "#F2E6C2",
            "fontFamily": "JosefinSans-Light",
            "fontSize": 10,
          }
        }
      >
        By continuing, you accept our
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#F2E6C2",
              "fontFamily": "JosefinSans-Light",
              "fontSize": 10,
              "fontWeight": "bold",
              "textDecorationLine": "underline",
            }
          }
        >
          Privacy Notices
        </Text>
        <Text
          style={
            Object {
              "color": "#F2E6C2",
              "fontFamily": "JosefinSans-Light",
              "fontSize": 10,
            }
          }
        >
           and 
        </Text>
        <Text
          style={
            Object {
              "color": "#F2E6C2",
              "fontFamily": "JosefinSans-Light",
              "fontSize": 10,
              "fontWeight": "bold",
              "textDecorationLine": "underline",
            }
          }
        >
          Terms of Use
        </Text>
      </View>
    </View>
  </View>
</View>
`;
